src/toolbar.js

```diff
@@ -1,179 +1,131 @@
-		{ type:'color', label:'Color', items:defaultColors },
-		{ type:'background', label:'Background color', items:defaultColors },
-		{ type:'link', label:'Link' }
-	]},
-
-	{ label:'Blocks', type:'group', items: [
-		{ type:'list', value:'bullet' },
-		{ type:'list', value:'ordered' }
-	]},
-
-	{ label:'Blocks', type:'group', items: [
-		{ type:'image', label:'Image' }
-	]}
-
-];
-
-var QuillToolbar = React.createClass({
-
-	displayName: 'Quill Toolbar',
-
-	propTypes: {
-		id:        T.string,
-		className: T.string,
-		items:     T.array
-	},
-
-	getDefaultProps: function(){
-		return {
-			items: defaultItems
-		};
-	},
-
-	renderGroup: function(item, key) {
-		return React.DOM.span({
-			key: item.label || key,
-			className:'ql-formats' },
-			item.items.map(this.renderItem)
-		);
-	},
-
-	renderChoiceItem: function(item, key) {
-		return React.DOM.option({
-			key: item.label || item.value || key,
-			value:item.value },
-			item.label
-		);
-	},
-
-	renderChoices: function(item, key) {
-		var attrs = {
-			key: item.label || key,
-			title: item.label,
-			className: 'ql-'+item.type
-		};
-		var self = this;
-		var choiceItems = item.items.map(function(item, key) {
-			if (item.selected) {
-				attrs.defaultValue = item.value;
-			}
-			return self.renderChoiceItem(item, key);
-		})
-		return React.DOM.select(attrs, choiceItems);
-	},
-
-	renderButton: function(item, key) {
-		return React.DOM.button({
-			type: 'button',
-			key: item.label || item.value || key,
-			value: item.value,
-			className: 'ql-'+item.type,
-			title: item.label },
-			item.children
-		);
-	},
-
-	renderAction: function(item, key) {
-		return React.DOM.span({
-			key: item.label || item.value || key,
-			className: 'ql-'+item.type,
-			title: item.label },
-			item.children
-		);
-	},
-
-	renderItem: function(item, key) {
-		switch (item.type) {
-			case 'group':
-				return this.renderGroup(item, key);
-			case 'font':
-			case 'header':
-			case 'align':
-			case 'size':
-			case 'color':
-			case 'background':
-				return this.renderChoices(item, key);
-			case 'bold':
-			case 'italic':
-			case 'underline':
-			case 'strike':
-			case 'link':
-			case 'list':
-			case 'bullet':
-			case 'ordered':
-			case 'indent':
-			case 'image':
-			case 'video':
-				return this.renderButton(item, key);
-			default:
-				return this.renderAction(item, key);
-		}
-	},
-
-	getClassName: function() {
-		return 'quill-toolbar ' + (this.props.className||'');
-	},
-
-	render: function() {
-		var children = this.props.items.map(this.renderItem);
-		var html = children.map(ReactDOMServer.renderToStaticMarkup).join('');
-		return React.DOM.div({
-			className: this.getClassName(),
-			style: this.props.style || {},
-			dangerouslySetInnerHTML: { __html:html }
-		});
-	}
-
-});
-
-module.exports = QuillToolbar;
-QuillToolbar.defaultItems = defaultItems;
-QuillToolbar.defaultColors = defaultColors;
+import React, { Component, PropTypes } from 'react'
```

Comment:
Please consider using separate [prop-types](https://github.com/facebook/prop-types) package instead using PropTypes from React. PropTypes from React will be deprecated in react@16.0.

Reply:
This PR is substantially out of date, but we are keeping it open as a reference for future  work. We have already moved to React 16-compatible utilities in the main repo.